### PR TITLE
Fix chained DC IST trigger

### PR DESCRIPTION
**Pull request by tnozicka**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1462131

Also related to https://github.com/openshift/origin/issues/14725#issuecomment-309746020

cc: @mfojtik @smarterclayton

### CODE REVIEW

**Comment by tnozicka** [comment]
[test]

**Comment by mfojtik** [comment]
can we add:

```golang
        if ref == p.LastTriggeredImage {
            continue
        }
```

to line 186? just to make sure.

**Comment by smarterclayton** [comment]
Add a test

**Comment by smarterclayton** [comment]
This may also be a problem with builds as well, since the other issue reported was builds

**Comment by tnozicka** [comment]
@mfojtik added the sanity check

**Comment by smarterclayton** [comment]
I'm looking at the build part right now, but I'd like to see a failing test that catches this.

**Comment by smarterclayton** [comment]
Follow up:

* we need an e2e test that verifies that a second image push results in a second deployment (the tests only verify the first deployment)

**Comment by tnozicka** [comment]
@smarterclayton added the unit [test]; 

follow up e2e is here: https://github.com/openshift/origin/issues/14797

**Comment by mfojtik** [comment]
[test]

**Comment by tnozicka** [comment]
re-[test]

**Comment by smarterclayton** [comment]
[severity:blocker]

File: pkg/image/trigger/deploymentconfigs/deploymentconfigs_test.go [hunk 1]
```diff
@@ -91,6 +91,28 @@ func TestDeploymentConfigReactor(t *testing.T) {
...
+			// no container, second run
+			tags: []fakeTagResponse{{Namespace: "other", Name: "stream-1:1", Ref: "image-lookup-2", RV: 3}},
+			obj: testDeploymentConfig([]deployapi.DeploymentTriggerImageChangeParams{
+				{
+					Automatic:          true,
+					From:               kapi.ObjectReference{Name: "stream-1:1", Namespace: "other", Kind: "ImageStreamTag"},
+					ContainerNames:     []string{"test"},
+					LastTriggeredImage: "image-lookup-1",
+				},
+			}, nil),
```
Comment (smarterclayton) [inline pkg/image/trigger/deploymentconfigs/deploymentconfigs_test.go]: you should create containers here as well that have the previous values, so you can verify they are also set properly.

**Comment by tnozicka** [comment]
[test]

**Comment by tnozicka** [comment]
[test]

File: pkg/image/trigger/deploymentconfigs/deploymentconfigs_test.go (hunk 1 above)
Comment (tnozicka) [inline pkg/image/trigger/deploymentconfigs/deploymentconfigs_test.go]: @smarterclayton I have extended the tests to verify substituting images as well. ptal

**Comment by smarterclayton** [comment]
Lgtm [severity:blocker][merge]

**Comment by smarterclayton** [comment]
Thanks!

**Comment by tnozicka** [comment]
re[test]

**Comment by mfojtik** [comment]
re-[merge]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/14043 [merge]

**Comment by smarterclayton** [comment]
[merge]

**Comment by deads2k** [comment]
re[test]

**Comment by smarterclayton** [comment]
@jupierce this is a stage blocker BTW

**Comment by smarterclayton** [comment]
[test]

**Comment by tnozicka** [comment]
[test] flake https://github.com/openshift/origin/issues/14898

**Comment by tnozicka** [comment]
[test] flake again (new-app) https://github.com/openshift/origin/issues/14043

**Comment by tnozicka** [comment]
flake https://github.com/openshift/origin/issues/14043 - second in a row

**Comment by tnozicka** [comment]
re-[test] (some weird networking/infra fail - https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_networking_minimal/3784/console)

**Comment by mfojtik** [comment]
[test] again

**Comment by eparis** [comment]
[merge] the network flake sure seems un-related.

**Comment by smarterclayton** [comment]
Flaked, merging

**Pull request merged**